docs/windows.md

```diff
@@ -0,0 +1,74 @@
+  - The build script located in each of the pacakge directories is paralleled by a build.ps1 script which is functionally similar.
+  - In the packages.windows tree there are setupinfo.json and setup.ps1 scripts wwhich provide setup information at package 
+    install time. Those files get included in the tar.xz build artifact.
+    
+- Changes to the python code in pkgpanda, release, gen, dcos_installer, ssh and related directories are
+  - is_windows variable so that all os-specific branches don't have to go through a string compare
+  - replacement of shell check_calls to if is_windows: check_call <windows> else: check_call <linux>
+  - added constants for paths so that the windows and linux paths need not require inline "if is_windows" sections.
+
+- A parallel set of yaml tmeplates.  These files are specialised by gen in the same manner as the windows version. The yaml
+  templates are used to create processed configuration files for a particular build and deployment. The windows versions will i
+  be pretty much line by line different owing to the difference in the two OS environments. For example, all paths will be
+  different, the syntax of environment setting is different, and windows has its own service manager making systemd units not
+  applicable to the windows build.  In place of systemd unit files, we will have a number of powershell scripts to implement
+  the same functionality so far as practical.
+
+## Packaging
+
+All of the components in DC/OS Windows are built into a single tar.xz file that eventually gets extracted to the linux equivelent path on Windows. Currently that path is %SystemDrive%:/DCOS rather than /opt/mesosphere, but that will be revised shortly. The built artifact is deployed from a separate bootstrap URL from the linux artifact. 
+
+
+## Building
+
+Like linux, the master artifact must be assembled somehow. Because the DC/OS build is made up of a changing list of components, the build tooling ends up looking like its own little package manager. Each component must be built from source, configured in a repeatable fashion and then added into the master artifact. The use of docker in building the windows artifact is similar todocker use in the linux build, however the docker container is setup very differently and must match the build of the Windows OS being used. 
+
+A DC/OS package is defined by two files. In windows, these are powershell rather than bash scripts, but their function is identical: [`build.ps1`][1] and [`buildinfo.json`][2]. These state what must be downloaded and how to build it. At build time, the toolchain takes care of building, packaging and including all the artifacts required into the master zip file.
+
+Because the dependent tools, packages, configuration, even formats are different between Windows and Linux, it is necessary to separately build the linux and windows packages on the respective systems.
```

Comment:
we would need to rearchitect the parser to support platform specific constructs  if this is the case. We can certainly do this merge going forwards, but to help us move forwards it would be good to go with the current solution and create an issue to rearchitect going forwards.

Reply:
Currently we are duplicating all buildinfos for windows due to some problems with the variant code, but if that is addressed we may be able to remove those which are the same.